src/modules/roc_audio/profiler.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2019 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+//! @file roc_audio/profiler.h
+//! @brief Profiler.
+
+#ifndef ROC_AUDIO_PROFILER_H_
+#define ROC_AUDIO_PROFILER_H_
+
+#include "roc_audio/frame.h"
+#include "roc_core/array.h"
+#include "roc_core/iallocator.h"
+#include "roc_core/list.h"
+#include "roc_core/noncopyable.h"
+#include "roc_core/rate_limiter.h"
+#include "roc_core/refcnt.h"
+#include "roc_core/time.h"
+#include "roc_packet/units.h"
+
+namespace roc {
+namespace audio {
+
+//! Profiler
+class Profiler : public core::NonCopyable<> {
```

Comment:
Could you please briefly describe our algorithm in the class comment?

Important things to note:

* Briefly state the goal: periodically report average processing speed (# of samples processed per time unit) during last N seconds.

* Briefly state the problem: we want to calculate SMA efficiently (with O(1) complexity, without allocations, and as lightweight as possible), but we have variable-size frames and SMA needs fixed-size chunks.

* Briefly describe the ring buffer, what does the chunk represent, and how its value is calculated.

* Briefly describe how we use weighted mean, SMA, and CMA.